mods/modernize-folder-picker-dialog.wh.cpp

```diff
@@ -0,0 +1,115 @@
+// ==WindhawkMod==
+// @id              modernize-folder-picker-dialog
+// @name            Modernize Folder Picker Dialog
+// @description     Replaces the classic "Browse For Folder" dialog
+// @version         1.0.0
+// @author          aubymori
+// @github          https://github.com/aubymori
+// @include         *
+// @compilerOptions -lole32
+// ==/WindhawkMod==
+
+// ==WindhawkModReadme==
+/*
+# Modernize Folder Picker Dialog
+
+This mod replaces the outdated, hard-to-use "Browse For Folder" dialog with a
+modern styled folder picker dialog, which is easier to navigate.
+
+**Before**:
+
+![Before](https://raw.githubusercontent.com/aubymori/images/main/modernize-folder-picker-dialog-before.png)
+
+**After**:
+
+![After](https://raw.githubusercontent.com/aubymori/images/main/modernize-folder-picker-dialog-after.png)
+*/
+// ==/WindhawkModReadme==
+
+#define RETURN_IF_FAILED(hr) if (FAILED(hr)) return SHBrowseForFolderW_orig(lpbi)
+
+#define E_CANCELLED (HRESULT)0x800704C7
+
+#include <initguid.h>
+#include <shlobj.h>
+#include <windhawk_utils.h>
+
+using SHBrowseForFolderW_t = decltype(&SHBrowseForFolderW);
+SHBrowseForFolderW_t SHBrowseForFolderW_orig = nullptr;
+PIDLIST_ABSOLUTE WINAPI SHBrowseForFolderW_hook(
+    LPBROWSEINFOW lpbi
+)
+{
+    if (!lpbi) return NULL;
+
+    /* Make the dialog */
+    IFileOpenDialog *pDialog = nullptr;
```

Comment:
This object is never freed. Same for `pRoot`, `pResult` and `pszPath` below.
You can use `winrt::com_ptr` to have all but `pszPath` automatically freed, for example:
https://github.com/ramensoftware/windhawk-mods/blob/16c1c93b12f2e2fb707c419efb19a7d3c3b80a11/mods/explorer-frame-classic.wh.cpp#L444

For `pszPath`, you need to call `CoTaskMemFree`, I don't think there's a smart object for it to free it automatically.